src/app/app.component.ts

```diff
@@ -1,41 +1,33 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
 import { Store } from '@ngrx/store';
-import * as ProductSelectors from './store/product/product.selectors';
-import * as ProductActions from './store/product/product.actions';
-import * as LangSelectors from './store/lang/lang.selectors';
-import * as LangActions from './store/lang/lang.actions';
 import { BehaviorSubject, Observable, Subject } from 'rxjs';
-import { Product } from './models/product';
-import { ProductService } from './services/product.service';
 import { finalize, map, switchMap, takeUntil, tap } from 'rxjs/operators';
 import { Lang } from './models/lang';
+import { Product } from './models/product';
+import { ProductService } from './services/product.service';
+import * as LangActions from './store/lang/lang.actions';
+import * as LangSelectors from './store/lang/lang.selectors';
+import * as ProductActions from './store/product/product.actions';
+import * as ProductSelectors from './store/product/product.selectors';
 
 @Component({
   selector: 'tk-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss'],
+  styleUrls: ['./app.component.scss']
 })
-export class AppComponent implements OnInit, OnDestroy {
-  products$: Observable<Array<Product>>;
-  search$: Observable<string>;
-  searchOptions$: Observable<Array<string>>;
-  lang$: Observable<Lang>;
+export class AppComponent implements OnDestroy {
+  readonly products$: Observable<Array<Product>> = this.store.select(ProductSelectors.products);
+  readonly search$: Observable<string> = this.store.select(ProductSelectors.search);
+  readonly searchOptions$: Observable<Array<string>> = this.products$.pipe(
+    map((products) => products.map((cv) => cv.name))
+  );
+  readonly lang$: Observable<Lang> = this.store.select(LangSelectors.langSelector);
   readonly loading$ = new BehaviorSubject<boolean>(true);
+  readonly ProductActions = ProductActions;
+  readonly LangActions = LangActions;
   private readonly destroy$ = new Subject<void>();
 
-  constructor(
-    private readonly store: Store,
-    private readonly productService: ProductService
-  ) {
-    this.lang$ = this.store.select(LangSelectors.langSelector);
-    this.products$ = this.store.select(ProductSelectors.products);
-    this.search$ = this.store.select(ProductSelectors.search);
-    this.searchOptions$ = this.products$.pipe(
-      map((products) => products.map((cv) => cv.name))
-    );
-  }
-
-  ngOnInit(): void {
+  constructor(readonly store: Store, private readonly productService: ProductService) {
```

Comment:
As far as I can tell, the product service is used only in the constructor. If it is true, then we don't need to make it a class member by marking it *private & readonly*

Reply:
why not? in such way we prevent access to it directly thru the component and it we ensure that it would not be accidentally overwritten inside the component constructor

Reply:
We don't need to keep a link inside of a component when it's not necessary. It's just performance improvement.
 
When it comes to access it directly from the constructor and rewriting its value is developers' and reviewers' job to avoid such cases.